### ISSUE TITLE
Fails to import spec files when jspm baseURL is not project root 

### ISSUE THREAD

**Issue by CxRes** [CanopyTax/node-jspm-jasmine#45]
I am using a directory structure as below:
```
package.json
src/
|- config.js
|- jspm_packages/
```

with `baseURL` set to `src` in `package.json`
```
"jspm": {
  "directories": {
    "baseURL": "src"
  },
  ...
```

Now when `System.import()` tries to fetch a file, it appends `src` (as it should) to the spec file path derived from Jasmine config. Consequently `System.import()` within `node-jspm-jasmine.js` fails to import the file (It sees the path `src/src/path/to/file` instead of `src/path/to/file`). 

Now, I am not sure as to the best way to resolve this. While `SystemJS.getConfig()` and `process.env` contain enough information to calculate this, it produces what looks like a rather contrived looking logic (though the benefit is that of no additional user configuration). Other way is that the user passes the `baseURL` when invoking `jspmjasmine` or `runTests()`. This is much simpler but does require additional user config.

I would like to hear your thoughts on how to solve this...

**Comment by internettrans** [comment]
Hey @CxRes, thanks for creating this issue -- I'm on vacation until Tuesday, but will take a look at this when I get back.

**Comment by CxRes** [comment]
No worries!!!!

**Comment by blittle** [comment]
@CxRes my thoughts are that `node-jspm-jasmine` tries to work out of the box with little to no config. If we could make it work without adding a configuration option, I think that would be the best way to go. PRs are definitely welcome as well!

**Comment by internettrans** [comment]
@CxRes Just spent some time looking into this, and yeah I think I see what you mean. It sounds like you're talking about [this SystemJS.import() call inside of node-jspm-jasmine](https://github.com/CanopyTax/node-jspm-jasmine/blob/master/src/node-jspm-jasmine.js#L359). In that code, SystemJS.import is being called with the file path that is found by `glob`ing the `spec_files` in the jasmine.json. But in the case you've described where there is a baseURL, you can't actually call SystemJS.import() with the file path because it always prepends everything with the baseURL.

Do you think it would be enough to do something like this?

```js
// replace line 359 of node-jspm-jasmine.js with this. https://github.com/CanopyTax/node-jspm-jasmine/blob/master/src/node-jspm-jasmine.js#L359
const baseUrl = SystemJS.getConfig().directories.baseURL
let pathToImport = file

if (pathToImport.startsWith(baseUrl)) {
  pathToImport = pathToImport.slice(baseUrl.length)
}

return SystemJS.import(pathToImport)
```

That approach doesn't use process.env -- how were you envisioning process.env being used?

I agree with @blittle that let's first start with trying to find a solution that doesn't require extra configuration, and only fall back to a configuration option if necessary. I believe my code sample above may fix the issue -- could you check if it works for you?

**Comment by CxRes** [comment]
@joeldenning The trouble is that `SystemJS.getConfig().directories.baseURL` is actually not defined. And then there is the trouble of two `baseURL` (in `package.json` and `config.js`) which can interact.  So you need to manually calculate this quantity:
```js
let sysPath = SystemJS.getConfig().baseURL.replace(/^file:\/\//, ''); //isolate the path from URL
sysPath = path.normalize(sysPath).replace(/^\\/, ''); //fixes for Windows

const baseUrl = path.relative(sysPath, process.cwd());
```

then in 359 this should work:
```js
let pathToImport = file;
if (pathToImport.startsWith(baseUrl)) {
  pathToImport = pathToImport.slice(baseUrl.length + 1); //remove the slash after the baseURL
}
```

You could also use `path.dirname(process.env.jspmConfigPath)` instead of `process.cwd()`. I am not sure if this has some actual benefit (except for looking good as it uses all jspm entities).

Also, there are two other places that use `SystemJS.import`. I am almost certain that helper files (line:262) will need this correction. I am not sure of the coverage bit (line:304) or why you are even adding `process.cwd()` to the import file path.

PS: I went back to a flat directory structure. However, if you are happy with my suggestions I could make a PR (since I did bring this up), although again I am not sure of the other two `SystemJS.import`. Or you could just copy the code above yourself :).

**Comment by internettrans** [comment]
@CxRes ah yeah that makes sense -- yes I'd be happy to review and merge a pull request doing what you've described. And yeah, I think we'd have to do it not only on line 359 but also for the helper files on line 262 and the coverage files on line 331. The coverage files are the files that the user cares about the coverage for, even if they were never imported by a test. It's to avoid saying you have "100% coverage" but have that really only mean "for the files that have tests" instead of "for all files."

I wasn't aware of `process.env.jspmConfigPath` -- that seems like it's probably more foolproof than `process.cwd()`, so let's switch to that.

**Comment by internettrans** [comment]
Fix released with node-jspm-jasmine@3.2.0
